libvast/builtins/operators/from_read.cpp

```diff
@@ -0,0 +1,257 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/string.hpp>
+#include <vast/concept/parseable/vast/identifier.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/error.hpp>
+#include <vast/legacy_pipeline.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <arrow/type.h>
+#include <caf/error.hpp>
+
+namespace vast::plugins::from_read {
+
+namespace {
+
+class from_operator final : public crtp_operator<from_operator> {
+public:
+  explicit from_operator(const loader_plugin& loader)
+    : loader_plugin_{&loader} {
+  }
+
+  explicit from_operator(const plugin& parser) : parser_plugin_{&parser} {
+  }
+
+  auto operator()(operator_control_plane& ctrl) const
+    -> caf::expected<generator<chunk_ptr>> {
+    if (loader_plugin_) {
+      VAST_ASSERT(!parser_plugin_);
+      auto loader = loader_plugin_->make_loader({}, ctrl);
+      if (!loader) {
+        return loader.error();
+      }
+      // TODO: Reconsider this.
+      return std::invoke(
+        [](loader_plugin::loader loader) -> generator<chunk_ptr> {
+          for (auto&& x : loader()) {
+            co_yield x;
+          }
+        },
+        std::move(*loader));
```

Comment:
```suggestion
      return loader_();
```

Does turning the `loader` into a field `loader_` and using it like in the suggestion work?

Reply:
I did something similar in my `write`/`to` branch and it seems to compile at least :thinking: